### PR TITLE
New term entry list.clear in dart

### LINKED ISSUE #4294: [Term Entry] Dart List .clear()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List .clear()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/clear/clear.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Vino-Siva**
### Description

New term entry for Dart - List.clear method

### Issue Solved

Closes #4294 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/list/terms/clear/clear.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+Title: 'Clear Method (List)'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:2]:
```suggestion
Title: 'clear()'
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  - 'Dart'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`clear()`** method in Dart is used to remove all elements from a List, leaving an empty list in its place. This method is handy when you need to reset or clear the contents of a List, making it ready for new data.
+
+## Implementation
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:18]:
```suggestion
## Syntax
```
This heading must be called Syntax.

File: content/dart/concepts/list/terms/clear/clear.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+```pseudo
+void clear();
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:21]:
```suggestion
listName.clear();
```
The syntax should be as minimal as possible and the users should understand it easily.

File: content/dart/concepts/list/terms/clear/clear.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+## Parameters
+
+This method does not take any parameters.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:26]:
```suggestion
```
We don't need this section.

File: content/dart/concepts/list/terms/clear/clear.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+## Return Type
+
+This method does not return any value.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:30]: This section is not required too but you can add this line in the introductory paragraph.

File: content/dart/concepts/list/terms/clear/clear.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+## Usage and Examples
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:32]:
```suggestion
## Example
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  print(growableList);
+  print(growableList.length); //length = 0
+  
+  final fixedList = List<int>.filled(3, 3, growable: false); //false is the default value for growable in list.filled constructor.
+  print(fixedList);
+  fixedList.clear();  //Throws an error for fixed list
+}
+```
+
+The Output is:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:54]:
```suggestion
The Output of the above code is:
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+[1, 1, 1]
+Uncaught Error: Unsupported operation: clear
+```
+
+> **Notes:**
+>
+>* The clear() method modifies the original list and does not create a new list.
+>* After calling clear(), the list will have a length of 0.
+>* If the list is already empty, calling clear() has no effect.
+>* If the list is a fixed list, calling clear() will throw an error.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:70]:
```suggestion
>* The `clear()` method modifies the original list and does not create a new list.
>* After calling `clear()`, the list will have a length of 0.
>* If the list is already empty, calling `clear()` has no effect.
>* If the list is a fixed list, calling `clear()` will throw an error.
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+The Output is:
+
+```shell
+[1, 1, 1]
+3
+[]
+0
+[1, 1, 1]
+Uncaught Error: Unsupported operation: clear
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:62]:
```suggestion
[1, 1, 1]
3
[]
0
[3, 3, 3]
Uncaught Error: Unsupported operation: clear
```
The output was slightly incorrect, I've edited it.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @Vino-Siva, excellent entry. I've suggested some changes, please make them.

**Comment by Vino-Siva** [comment]
> Hey @Vino-Siva, excellent entry. I've suggested some changes, please make them.

Thanks 😄 ... Made the changes you have suggested.

File: content/dart/concepts/list/terms/clear/clear.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  print(growableList);
+  print(growableList.length); //length = 0
+  
+  final fixedList = List<int>.filled(3, 3, growable: false); //false is the default value for growable in list.filled constructor.
+  print(fixedList);
+  fixedList.clear();  //Throws an error for fixed list
+}
+```
+
+The Output of the above code is:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:48]:
```suggestion
The output of the above code is:
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
+---
+Title: 'clear()'
+Description: 'Removes all objects from the list; the length of the list becomes zero.'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:3]:
```suggestion
Description: 'Removes all objects from the list and the length of the list becomes zero.'
```

File: content/dart/concepts/list/terms/clear/clear.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  - 'Dart'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`clear()`** method in Dart is used to remove all elements from a List, leaving an empty list in its place. This method is handy when you need to reset or clear the contents of a List, making it ready for new data.
+
+This method does not return any value.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/clear/clear.md:18]:
```suggestion
The **`clear()`** method in Dart is used to remove all elements from a List, leaving an empty list in its place. This method is handy when you need to reset or clear the contents of a List, making it ready for new data. This method does not return any value.
```

**Review by avdhoottt — APPROVED** [review]
Great!! Look good for the second review.

**Review by Sriparno08 — APPROVED** [review]
Looks good 👍

**Pull request merged**